frame/treasury-oracle/src/lib.rs

```diff
@@ -0,0 +1,249 @@
+//! ### Permissioned Functions
+//!
+//! * `create`: Creates a new asset conversion rate.
+//! * `remove`: Removes an existing asset conversion rate.
+//! * `update`: Overwrites an existing assert conversion rate.
+//!
+//! Please refer to the [`Call`] enum and its associated variants for documentation on each
+//! function.
+//!
+//! ### Assumptions
+//!
+//! * Conversion rates will not be used to determine the payment amount in another asset.
+//! * Conversion rates will be used to determine the tier of the spender status, e.g.
+//!   `SmallSpender`, `MediumSpender`, `BigSpender` or `Treasurer`.
+//! * All conversion rates reflect the ration of some asset to native, e.g. native = asset * rate.
+//!
+//! ## Related Modules
+//! * [`Treasury`](../treasury/index.html)
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::traits::{
+	fungible::Inspect,
+	tokens::{Balance, BalanceConversion},
+};
+use sp_runtime::{traits::Zero, FixedPointNumber, FixedPointOperand, FixedU128};
+
+pub use pallet::*;
+pub use weights::WeightInfo;
+
+#[cfg(feature = "runtime-benchmarks")]
+pub mod benchmarking;
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+pub mod weights;
+
+// Type alias for `frame_system`'s account id.
+type AccountIdOf<T> = <T as frame_system::Config>::AccountId;
+// This pallet's asset id and balance type.
+type AssetIdOf<T> = <T as Config>::AssetId;
+// Generic fungible balance type.
+type BalanceOf<T> = <<T as Config>::Currency as Inspect<AccountIdOf<T>>>::Balance;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// The Weight information for extrinsics in this pallet.
+		type WeightInfo: WeightInfo;
+
+		/// The runtime event type.
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+		/// The origin permissioned to create a conversion rate for an asset.
+		type CreateOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+
+		/// The origin permissioned to remove an existing conversion rate for an asset.
+		type RemoveOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+
+		/// The origin permissioned to update an existiing conversion rate for an asset.
+		type UpdateOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+
+		/// The units in which we record balances.
+		type Balance: Balance + FixedPointOperand;
+
+		/// The currency mechanism for this pallet.
+		type Currency: Inspect<Self::AccountId, Balance = Self::Balance>;
+
+		/// The identifier for the class of asset.
+		type AssetId: Member + Parameter + Copy + MaybeSerializeDeserialize + MaxEncodedLen;
+	}
+
+	/// Maps an asset to its fixed point representation in the native balance.
+	///
+	/// E.g. `native_amount = asset_amount * ConversionRateToNative::<T>::get(asset_id)`
+	#[pallet::storage]
+	#[pallet::getter(fn conversion_rate_to_native)]
```

Comment:
We are getting rid of storage getters https://github.com/paritytech/substrate/pull/13365